Spider.php

```diff
@@ -70,6 +75,59 @@ public function getRequestHandler()
     }
 
     /**
+     * Sets the blacklist
+     *
+     * @argument array
+     */
+    public function setBlacklist(array $blacklist){
+	    $this->blacklist = $blacklist;
+    }
+
+    /**
+     * Returns the blacklisted url's
+     *
+     * @return array
+     */
+    public function getBlacklist(){
+    	return $this->blacklist;
+    }
+
+    /**
+     * Checks is a URL is blacklister
+     *
+     * @return bool
+     */
+    public function isUrlBlacklisted(Uri $uri)
+    {
+        $isBlacklisted = false;
+        
+        if (in_array($uri->toString(), $this->blacklist)) {
+            return true;
+        }
+
+        array_walk($this->blacklist, function($blacklistUrl) use ($uri, &$isBlacklisted) {
```

Comment:
I think $blacklistUrl should be renamed to $blacklistPattern, since in this scope you are expecting it to be a pattern.